src/cpp/database/database.cpp

```diff
@@ -533,18 +533,31 @@ void Database::insert_nts(
         }
         case DataKind::NETWORK_LATENCY:
         {
-            const NetworkLatencySample& network_latency = dynamic_cast<const NetworkLatencySample&>(sample);
-
-            // Create locator if it does not exist
-            std::shared_ptr<Locator> locator = get_locator_nts(entity_id);
-
-            // Create remote_locator if it does not exist
-            get_locator_nts(network_latency.remote_locator);
-
-            // Add the info to the locator
-            locator->data.network_latency_per_locator[network_latency.remote_locator].push_back(network_latency);
+            /* Check that the entity is a known locator */
+            auto locator_it = locators_.find(entity_id);
+            if (locator_it != locators_.end())
+            {
+                const NetworkLatencySample& network_latency = dynamic_cast<const NetworkLatencySample&>(sample);
+                /* Check that the remote entity is a known locator */
+                if (locators_.find(network_latency.remote_locator) != locators_.end())
+                {
+                    locator_it->second->data.network_latency_per_locator[network_latency.remote_locator].push_back(
+                        network_latency);
+                }
+                else
+                {
+                    throw BadParameter(std::to_string(
+                                      network_latency.remote_locator.value()) +
+                                  " does not refer to a known locator");
```

Comment:
We are calling this method only on our internal code from the queue. And we are already creating the locators in the queue if they do not exists. I'd change this into an assert, as it would mean there is an error in our implementation.

Reply:
Because of this, I also modified database_test to make `ASSERT_DEATH` instead of `ASSERT_THROW`